File: src/problem/leetcode/easy/PascalsTriangle.kt
package problem.leetcode.easy

class PascalsTriangle {
    fun generate(numRows: Int): List<List<Int>> {
        val arr = Array(numRows) { i -> IntArray(i + 1) }
        arr[0][0] = 1

        for (i in 0 until numRows - 1) {
            for (j in 0..i) {
                arr[i+1][j] += arr[i][j]
                arr[i+1][j+1] += arr[i][j]
            }
        }

        return arr.map { it.toList() }
    }
}

File: src/problem/leetcode/easy/pascals-triangle.py
class PascalsTriangle:
    def generate(self, numRows: int) -> list[list[int]]:
        if numRows == 0:
            return []

        res = [[1]]

        for i in range(1, numRows):
            prev_row = res[-1]

            new_row = [1] + [prev_row[j] + prev_row[j+1] for j in range(len(prev_row) - 1)] + [1]

            res.append(new_row)

        return res